src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs

```diff
@@ -193,6 +200,84 @@ private List<string> ResolveByExternalReferencePackages(List<string> uidList, Co
             return list;
         }
 
+        private async Task<List<string>> ResolveByXRefServiceAsync(List<string> uidList, ConcurrentDictionary<string, XRefSpec> externalXRefSpec)
+        {
+            if (XRefServiceUrls == null || XRefServiceUrls.Length == 0)
+            {
+                return uidList;
+            }
+            string requestUrl = XRefServiceUrls[0];
+            var list = new List<string>();
+            int pieceSize = 1000;
+            using (var client = new HttpClient())
+            {   
+                try
+                {
+                    client.BaseAddress = new Uri(requestUrl);
+                }
+                catch (UriFormatException e)
+                {
+                    Logger.LogWarning($"Ignore invalid url: {requestUrl}." + e.Message);
+                    return uidList;
+                }
+                catch (ArgumentException e)
+                {
+                    Logger.LogWarning($"Ignore invalid url: {requestUrl}." + e.Message);
+                    return uidList;
+                }
+                
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+
+                for (int i = 0; i < uidList.Count; i += pieceSize)
+                {
+                    List<string> smallPiece;
+                    smallPiece = uidList.GetRange(i, Math.Min(pieceSize, uidList.Count - i));
+
+                    StringContent content = new StringContent(JsonUtility.Serialize(smallPiece), System.Text.Encoding.UTF8, "application/json");
+                    HttpResponseMessage response = null;
+                    try
+                    {
+                        response = await client.PostAsync("", content);
+                    }
+                    catch (HttpRequestException e)
+                    {
+                        Logger.LogWarning(e.InnerException.Message + "\n" + smallPiece.Count + " uids being resolved failed, for example including " + smallPiece[0]);
```

Comment:
Failed to resolve smallPiece.Count uids from url: smallPiece.Take(10).ToDelimitedString() ...

---
In reply to: [126866203](https://github.com/dotnet/docfx/pull/1837#discussion_r126866203) [](ancestors = 126866203)